crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -798,9 +811,11 @@ mod tests {
         let faucet_address = faucet.wallet_mut().active_address().unwrap();
         let uuid = Uuid::new_v4();
         let coin_response = faucet.prepare_gas_coin(100, uuid).await;
+        let coin_id;
 
         match coin_response {
             GasCoinResponse::ValidGasCoin(gas) => {
+                coin_id = gas;
```

Comment:
We can restructure this code with my favourite code review suggestion of all time: `let ... else`.  It avoids the nesting, and the need to create an uninitialised binding, and then set it within the `match`.

```rust
let GasCoinResponse::ValidGasCoin(coin_id) = faucet.prepare_gas_coin(100, uuid).await else {
    panic!("prepare_gas_coin did not give a valid coin.")
};

let tx_data = faucet
    .build_pay_sui_txn(coin_id, faucet_address, recipient, &[100], 200_000_000)
    .await
    .map_err(FaucetError::internal)
    .unwrap();

let mut wal = faucet.wal.lock().await;
// Check no wal
assert!(wal.log.is_empty());
wal.reserve(Uuid::new_v4(), coin_id, recipient, tx_data)
    .map_err(FaucetError::internal)
    .ok();
// Check the wal has one entry
assert!(wal.log.is_empty());
drop(wal);
```